File: ServerNetty/src/gege/game/GameItem.java
package gege.game;

public class GameItem extends GameEntity {
	private static int _nextId = 0;
	
	
	public static int TYPE_RED = 0;
	public static int TYPE_BLUE = 1;
	
	
	// 唯一标识
	final private int m_id;
	
	
	private int m_type;
	
	
	public GameItem(int type) {
		m_id = _nextId++;
		m_type = type;
	}
	
	
	public int getId(){
		return m_id;
	}
	
	
	public int getType(){
		return m_type;
	}
	
	
	public boolean inPos(Vector3 v){
		if(v.x == x && v.y == y)
			return true;
		
		return false;
	}
	

	@Override
	public void onUpdate() {
	}

}

File: Assets/Scripts/Entity/GameItem.cs
using UnityEngine;
using System.Collections;

public class GameItem : GameEntity {


    public const int TYPE_RED = 0;
    public const int TYPE_BLUE = 1;


    int m_type = -1;
    int m_id;

    public GameItem(string prefab, int type, int id, float x, float y)
        : base(prefab)
    {
        m_type = type;
        m_id = id;
        setPosition(x, y);
    }


    public int getId()
    {
        return m_id;
    }


    public override void onUpdate()
    {
    }
}
